django/middleware/csrf.py

```diff
@@ -221,7 +271,15 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
                 # branches that call reject().
                 return self._accept(request)
 
-            if request.is_secure():
+            # Reject the request if the origin header doesn't match an allowed
+            # value.
+            origin_verified = False
+            if 'HTTP_ORIGIN' in request.META:
+                if not self._origin_verified(request):
+                    return self._reject(request, REASON_BAD_ORIGIN % request.META['HTTP_ORIGIN'])
+                origin_verified = True
+
+            if request.is_secure() and not origin_verified:
```

Comment:
If I read correctly `origin_verified` isn't used anywhere else so we can simplify this to:
```suggestion
            # Reject the request if the origin header doesn't match an allowed
            # value.
            if 'HTTP_ORIGIN' in request.META and not self._origin_verified(request):
                return self._reject(request, REASON_BAD_ORIGIN % request.META['HTTP_ORIGIN'])

            if request.is_secure():
```

Reply:
The suggested change doesn't pass the tests because the purpose of `origin_verified` is to skip referer checking if origin checking passes.

Reply:
Right, my bad.

Reply:
I removed `origin_verified` by using an `elif`. I think it's a readability improvement.

Reply:
But now it will never fall back to the referer check if the Origin header exists (no matter if the origin header validated fine or not)…

Reply:
It didn't fallback before. If the origin check fails, the request is rejected.

Reply:
Arg true, doing to many things at the same time. Yes the `elif` is vastly better than the extra variable before. :+1: